cron.c

```diff
@@ -304,16 +306,19 @@ find_jobs(int vtime, cron_db *db, int doWild, int doNonWild) {
 	 */
 	for (u = db->head; u != NULL; u = u->next) {
 		for (e = u->crontab; e != NULL; e = e->next) {
+			bool lastdom = (e->flags & DOM_LAST) != 0 &&
+				tom.tm_mday == 1;
+			bool thisdom = lastdom || bit_test(e->dom, dom) != 0;
+			bool thisdow = bit_test(e->dow, dow);
+
 			Debug(DSCH|DEXT, ("user [%s:%ld:%ld:...] cmd=\"%s\"\n",
 			    e->pwd->pw_name, (long)e->pwd->pw_uid,
 			    (long)e->pwd->pw_gid, e->cmd))
 			if (bit_test(e->minute, minute) &&
 			    bit_test(e->hour, hour) &&
 			    bit_test(e->month, month) &&
-			    ( ((e->flags & DOM_STAR) || (e->flags & DOW_STAR))
-			      ? (bit_test(e->dow,dow) && bit_test(e->dom,dom))
-			      : (bit_test(e->dow,dow) || bit_test(e->dom,dom))
-			    )
+			    ((e->flags & (DOM_STAR|DOW_STAR)) != 0
+			     ? (thisdom && thisdow) : (thisdow || thisdow))
```

Comment:
I think you meant `thisdom || thisdow`?

Reply:
Indeed, it's really easy to came across this kind of typo.

Reply:
ok.